.github/workflows/java-publish.yml

```diff
@@ -0,0 +1,82 @@
+name: Build and publish Java packages
+on:
+  release:
+    types: [released]
+  pull_request:
+    paths:
+      - .github/workflows/java-publish.yml
+
+jobs:
+  # macos-arm64:
+  #   name: Build on MacOS Arm64
+  #   runs-on: macos-14
+  #   timeout-minutes: 80
+  #   defaults:
+  #     run:
+  #       working-directory: ./java
+  #   steps:
+  #     - name: Checkout repository
+  #       uses: actions/checkout@v4
+  #     - uses: Swatinem/rust-cache@v2
+  #     - name: Install dependencies
+  #       run: |
+  #         brew install protobuf
+  #     - name: Build release
+  #       run: |
+  #         cargo build --release
+  #     - uses: actions/upload-artifact@v4
+  #       with:
+  #         name: liblancedb_jni_darwin_aarch64.zip
+  #         path: target/release/liblancedb_jni.dylib
+  #         retention-days: 1
+  #         if-no-files-found: error
+  linux-arm64:
+    name: Build on Linux Arm64
+    runs-on: warp-ubuntu-2204-arm64-8x
+    timeout-minutes: 80
+    defaults:
+      run:
+        working-directory: ./java
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          toolchain: "1.79.0"
+          cache-workspaces: "src/rust"
+          # Disable full debug symbol generation to speed up CI build and keep memory down
+          # "1" means line tables only, which is useful for panic tracebacks.
+          rustflags: "-C debuginfo=1"
+      - name: Install dependencies
+        run: |
+          sudo apt -y -qq update
+          sudo apt install -y protobuf-compiler libssl-dev pkg-config
+      - name: Build release
```

Comment:
looks like we need to do something similar to https://github.com/lancedb/lance/pull/2613/files#r1720849909 before the Build release in order to incorporate the right GLIBC